lib/src/darwin.dart

```diff
@@ -404,20 +434,28 @@ class AVAudioSessionInterruptionOptions {
   int get hashCode => value.hashCode;
 }
 
-///// The route change reasons for [AVAudioSession].
-//enum AVAudioSessionRouteChangeReason {
-//  unknown,
-//  newDeviceAvailable,
-//  oldDeviceUnavailable,
-//  categoryChange,
-//  override,
-//  wakeFromSleep,
-//  noSuitableRouteForCategory,
-//  routeConfigurationChange,
-//}
-//
-//enum AVAudioSessionSilenceSecondaryAudioHintType { begin, end }
-//
+class AVAudioSessionRouteChange {
+  final AVAudioSessionRouteChangeReason reason;
+  // add everything else later
+
+  AVAudioSessionRouteChange({@required this.reason});
+}
+
+/// The route change reasons for [AVAudioSession].
+enum AVAudioSessionRouteChangeReason {
+  unknown,
+  newDeviceAvailable,
+  oldDeviceUnavailable,
+  categoryChange,
+  override,
+  wakeFromSleep,
+  noSuitableRouteForCategory,
+  routeConfigurationChange,
+}
+
+/// The interruption types for [AVAudioSessionSilenceSecondaryAudioHint].
+enum AVAudioSessionSilenceSecondaryAudioHintType { began, end }
```

Comment:
When I look at the definitions of the constants, they are "begin = 1" and "end = 0", so to get the right numerical values (and names) it needs to be:

```dart
enum AVAudioSessionSilenceSecondaryAudioHintType { end, begin }
```